index.js

```diff
@@ -79,10 +76,26 @@ server.post('/api/users', (req,res) => {
 server.delete('/api/users/:id', (req,res) => {
   db.remove(req.params.id)
     .then( numDel => {
-      if (numDel === 0) res.status(404).send('ID not found.. try again')
+      if (numDel === 0) res.status(404).json({message: "The user with the specified ID does not exist." })
       if (numDel > 0) res.status(200).send(`${numDel} user(s) was deleted `)
     }, 
-    reason => {res.status(500)});
+    reason => {res.status(500).json({error: "The user could not be removed"})});
+})
+
+server.put('/api/users/:id', (req,res) => {
+  let updateUser = req.body
+
+  if (!updateUser.hasOwnProperty('name') || !updateUser.hasOwnProperty('bio'))
```

Comment:
Interesting use of `hasOwnProperty()`. Also, does the user get created with dates automatically? I put dates in via Date.now(), but looks like it does it by default. If so, yours is more succinct! Here's my put method:
```
server.put("/api/users/:id", (req, res) => {
  const { id } = req.params;
  const user = req.body;
  const dates = {
    created_at: Date.now(),
    updated_at: Date.now(),
  }
  const updateUser = db.update(id, {...user, ...dates});

  if (!req.body.name || !req.body.bio) {
    return res
      .status(400)
      .json({ errorMessage: "Please provide name and bio for the user." });
  }

  updateUser
    .then(updatedUserCount => {
      if (updatedUserCount !== 1) {
        return res
          .status(404)
          .json({ errorMessage: "The user with the specified ID does not exist." });
      }
      res.status(200).json({...user, ...dates});
    })
    .catch(err => {
      res
      .status(500).json({ errorMessage: "The user information could not be modified.", err: err })
      .end();
    });
});
```
Update: Also now that I've just re-looked at my code, I'll need to remove the `created_at: Date.now()`, because it shouldn't be changing the created date! 😄